OVERVIEW.md

```diff
@@ -0,0 +1,180 @@
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `NOT_BLANK`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+ 
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `REGEX`
+
+For `STRING` columns. This rule type allows the specification of a Python regular expression to define allowed patterns.
+
+The example shows a check for a valid email address.
+
+The rule is applied to a column in a table through a rule binding.
+
+```yaml
+rules:
+  VALID_EMAIL:
+    rule_type: REGEX
+    dimension: conformance
+    params:
+      pattern: |-
+        ^[^@]+[@]{1}[^@]+$
+
+rule_bindings:
+  VALID_EMAIL:
+    entity_id: TEST_DATA
+    column_id: EMAIL
+    row_filter_id: NONE
+    rule_ids:
+      - VALID_EMAIL
+```
+
+> Complete example TODO
+
+#### Rule Type `CUSTOM_SQL_EXPRESSION`
+
+This rule type allows the specification of a condition using SQL (as in a WHERE clause). The validation fails when the condition is violated.
+
+The example condition performs a SELECT on a reference table to compare to a list of known currencies.
+
+The SQL condition references a parameter $column, which is set through the rule binding.
+
+```yaml
+rules:
+  CORRECT_CURRENCY_CODE:
+    rule_type: CUSTOM_SQL_EXPR
+    dimension: conformance
+    params:
+      custom_sql_expr: |-
+        $column in (select distinct currency_code from `<reference_dataset-id>.currency_codes`)
+
+rule_bindings:
+  LOCATION_LAT:
+    entity_id: TEST_DATA
+    column_id: CURRENCY
+    row_filter_id: NONE
+    rule_ids:
+      - CORRECT_CURRENCY_CODE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `CUSTOM_SQL_STATEMENT`
+
+The “custom SQL statement” allows you to specify a complete SQL statement. Any rows selected by the statement are considered violations of the rule.
+
+The SQL statement references the source data as “data”. The dataset is defined through the entity_id field in the rule binding.
+
+The SQL statement can also be parametrized using custom SQL arguments as shown in the example. The example provides a way to check the range of values in a column. 
+
+This rule type is intended for set-level validation: To check properties of a set, such as the total row count, or the standard deviation of a column. Therefore, while we reported the number of rows returned as the number of failures, the reporting varies from the other rule types.
```

Comment:
pls bring this parameter closer to the start of the section as it's quite important